### DISCUSSION TITLE
How to deals with lot of test +300

### DISCUSSION THREAD

**Discussion by Horsty80** [codeceptjs/CodeceptJS#4210] [General]
I'm asking how you deals with a package of e2e test like +300 tests.

This is the actual amount of test that i maintain. 
I have 2 approach on how to launch them.

First - All test in 8 workers - that we call `all job` & duration of ~15min
`codeceptjs run-workers 8 --plugins allure`

Second - 30 jenkins jobs that run only 10 tests with grep option - that we call `split job` & duration of ~2/3min per job
`codeceptjs run-workers 2 --plugins allure --grep`

Depending on which approach i use, i didn't get the same result and the same % of green run.

With the second `split job` i have a score of +95% of green run that is awesome.
No flaky, no failed job, no timeout, everything is ok.
The only issue is that all split job don't know the result of other so i can not get an aggregate result (except by go to the page and check visually all green marker)

With the first `all job` it's more complicated. I have ~10% of green run, all other are red or timeout.
The main issue is that the run are stuck i don't know why and then timeout.
Sometimes, 299 test are green and only 1 are red because browser has not starting 
The guilty test is not the same and change everytime.
It's like, when i get more and more test, try to rening each in the same job cause issue. I don't know how to explain that.

Have you experimented same thing ?

**Comment by adudyak** [comment]
Do they work fine, when executed with 1 worker?
As bad solution, can suggest https://codecept.io/basics/#retry-scenario

**Reply by Horsty80** [reply]
I will try tomorrow, but it will be very long run ^^ not the target

I have already add retry on Scénario / Feature / Before between 3 and 1
